index.js

```diff
@@ -234,6 +261,28 @@ function checkRootPathForErrors (fastify, rootPath) {
   }
 }
 
+function checkEncodingHeaders (headers, checked) {
+  if (!('accept-encoding' in headers)) return
+
+  let ext
+  const accepted = headers['accept-encoding'].split(', ')
```

Comment:
Would this mean also pulling in `encoding-negotiator` as a dependency?

Reply:
The referenced module is quite small, but I don't think that's necessary as most browsers just send the list without priority. For chrome and safari it's `accept-encoding: gzip, deflate, br`

Reply:
> Would this mean also pulling in encoding-negotiator as a dependency?

Yes

I agree that it could seems to much, but I think there is only 2 sentence here:
- support the standard
- not support the standard

To avoid issue, as pointed out, we should in every case add some code like the regxep match, but it seems easier adding the `encoding-negotiator` that is battle-tested for this purpose

(sorry for the late reply)